File: Producto.java
public class Producto {
    private static int nextID = 0;
    int id;
    int stock;
    String nombre;
    int numSerie;
    double precio;
    String fechaImportacion;

    public Producto(String nombre, int numSerie, double price, String fechaImportacion, int stock) {
        this.nombre = nombre;
        this.numSerie = numSerie;
        this.precio = price;
        this.fechaImportacion = fechaImportacion;
        this.stock = stock;
        this.id = nextID;
        nextID++;
    }

    protected int getID(){
        return id;
    }

    protected String getNombre(){
        return nombre;
    }

    protected boolean agregarStock(int stock){
        boolean agregado = false;
        if(stock>0){
            this.stock+=stock;
            agregado = true;
        }
        return agregado;
    }

    protected boolean quitarStock(int stock){
        boolean quitado = false;
        if(!(stock<0 || stock>this.stock)){
            this.stock-=stock;
            quitado = true;
        }
        return quitado;
    }

    protected int getStock(){
        return stock;
    }

    public double getPrecio(){
        return precio;
    }

}

File: Producto.py
class Producto:
    nextID = 0

    def __init__(self, nombre, numSerie, price, fechaImportacion, stock):
        self.nombre = nombre
        self.numSerie = numSerie
        self.precio = price
        self.fechaImportacion = fechaImportacion
        self.stock = stock
        self.id = Producto.nextID
        Producto.nextID += 1

    def getID(self):
        return self.id

    def getNombre(self):
        return self.nombre

    def agregarStock(self, stock):
        agregado = False
        if stock > 0:
            self.stock += stock
            agregado = True
        return agregado

    def quitarStock(self, stock):
        quitado = False
        if not (stock < 0 or stock > self.stock):
            self.stock -= stock
            quitado = True
        return quitado

    def getStock(self):
        return self.stock

    def getPrecio(self):
        return self.precio